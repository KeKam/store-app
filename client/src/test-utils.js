import React from 'react';
import { render as rtlRender } from '@testing-library/react';
import { createStore, applyMiddleware } from 'redux';
import { Provider } from 'react-redux';

import rootReducer from './redux/root-reducer';

// const render = (
//   component,
//   {
//     initialState,
//     store = createStore(rootReducer, initialState),
//     ...renderOptions
//   } = {}
// ) => {
//   return {
//     ...rtlRender(<Provider store={store}>{component}</Provider>),
//     store,
//   };
//   // const Wrapper = ({ children }) => {
//   //   return <Provider store={store}>{children}</Provider>;
//   // };
//   // return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
// };

const render = (component, { initialState } = {}) => {
  const actions = [];
  const observerMiddleware = () => (next) => (action) => {
    actions.push(action);
    return next(action);
  };
  const store = createStore(
    rootReducer,
    initialState,
    applyMiddleware(observerMiddleware)
  );
  const utils = {
    dispatch(action) {
      return store.dispatch(action);
    },
    getDispatchedActions() {
      return actions;
    },
    getState() {
      return store.getState();
    },
  };
  return {
    ...rtlRender(<Provider store={store}>{component}</Provider>),
    ...utils,
  };
};

// re-export everything
export * from '@testing-library/react';

// override render method
export { render };
