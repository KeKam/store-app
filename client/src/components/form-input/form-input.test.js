import React from 'react';
import '@testing-library/jest-dom';

import { render, screen } from '../../test-utils';
import FormInput from './form-input';

describe('FormInput component', () => {
  it('should render FormInput component', () => {
    const mockInputProps = {
      label: 'Email',
      value: 'testUser@gmail.com',
      onChange: jest.fn(),
    };
    const { asFragment } = render(<FormInput {...mockInputProps} />);

    expect(asFragment()).toMatchSnapshot();
    expect(screen.getByText('Email')).toBeInTheDocument();
    expect(screen.getByRole('textbox', { name: '' })).toHaveValue(
      'testUser@gmail.com'
    );
  });
});
