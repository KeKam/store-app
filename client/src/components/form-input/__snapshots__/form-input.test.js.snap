// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormInput component should render FormInput component 1`] = `
<DocumentFragment>
  <div
    class="sc-bdVaJa eEzrUY"
  >
    <input
      class="sc-bwzfXH eKtSpl"
      value="testUser@gmail.com"
    />
    <label
      class="sc-htpNat cAoTjx"
      value="testUser@gmail.com"
    >
      Email
    </label>
  </div>
</DocumentFragment>
`;
